### ISSUE TITLE
French dictionnary update

### ISSUE THREAD

**Issue by breversa** [AnySoftKeyboard/LanguagePack#12]
Hi everyone,

Several people (and myself) who've tried the french language pack seem rather disappointed by its dictionnary correction/prediction system. For instance, it fails to recognize many conjugated verb forms, even common ones. That actually led me to switch to SwiftKey for my everyday phone, though I keep using ASK for my other devices.

Anyway : I believe the french dictionnary could be updated/upgraded with data from dicollecte.org (http://dicollecte.org/home.php?prj=fr), which provides (free-as-in-beer) french dictionnaries and thesauruses.

I already wrote about it there : https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/109, but I guess a new issue here would be more logical.

**Comment by breversa** [comment]
So, here's what I did so far :
* I took the latest lexicon data (not the dictionnary for Hunspell) from dicollecte : http://dicollecte.org/download.php?prj=fr
* I trimmed the above file from its text and field headers
* I put it through a tiny AWK script I wrote (https://my.owndrive.com/index.php/s/3486b43d01e665b80f25c9d62f7f1007) to generate a fr.xml file that lists words and their frequency index (https://my.owndrive.com/index.php/s/8e5c75acbf7e8766b2eb6efb09d24fa7)
… and that's it.

The file includes more than 264 000 words ordered by a frequency index raging from 9 (most frequent) to 1 (less frequent). Not all these are needed, so I guess words with a frequency of 1 or 2 could be safely ignored.

What is now needed now is to integrate that fr.xml file into the french language pack for ASK, but that's what I don't know how.
 
So if anyone could either :
* update the current french pack with my fr.xml file
OR
* tell me how to do it myself,
that'd be awesome. :-)

**Comment by xavihernandez** [comment]
+1

Would be really good. As apostoph is really badly handled. Conjugation missing, etc.

**Comment by breversa** [comment]
I can remake/update the fr.xml file as needed, but I don't know how to make it into a language pack. :-(

**Comment by xavihernandez** [comment]
I don't know neither...

**Comment by xavihernandez** [comment]
No one can help here?
@menny any clue?

**Comment by ludovig** [comment]
Hi,

I just started to use your french dictionary today in my own any soft keyboard package (https://github.com/ludovig/vimboard/tree/master/android-vimboard-ask).
To convert it to binary I found some documentation in that old wiki page, https://code.google.com/archive/p/softkeyboard/wikis/BinaryDictionaries.wiki
The necessary script, makedict_Linux is in the src of the archive of this google code page. I dunno if there is some more recent version and easily accessible of this script elsewhere.

**Comment by menny** [comment]
There was a lot of development in Language Packs.
Check out https://github.com/AnySoftKeyboard/LanguagePack/tree/Hebrew for example.
Try to convert your project to that structure.

**Comment by breversa** [comment]
Shalom @menny 
I'm happy that you took the time to reply to that issue. However, could you please be a little more specific with your advice ? What file(s ?) do I need to look at ? Words.xml, maybe ?

I had a look at the other files, but unfortunately, I can't read hebrew (well… I tried to learn, but years ago !). So any help is welcome ! :-)

(To be honest, I've switched to Swiftkey several months ago because their dictonnary was way better, but I'd be more than happy to work on bringing ASK back to the same level and switch again. :-) )

**Comment by menny** [comment]
@breversa @ludovig , if you look at the latest _master_ of _LanguagePack_, you'll see that now it has been converted to use _Gradle_, also, it has a host of utility tasks that can convert HTML pages, TEXT files and standard `words.xml` files into binary dictionaries.

The Lithuanian branch(https://github.com/AnySoftKeyboard/LanguagePack/tree/Lithuanian) shows a very good implementation of a language pack.

**Comment by breversa** [comment]
@ludovig : do you want me to update my dictonnary with the latest data from dicollecte.org (check it out, since you're french-speaking : they have awesome projects !) so you can (try to) build a proper french language pack ? Or is my AWK script above enough for you ? :-)

**Comment by friesenkiwi** [comment]
@breversa @ludovig @xavihernandez we resurrected the old sources for the French pack from the SVN archive, modernized the build and replaced the dictionary by the one from the AOSP sources: https://android.googlesource.com/platform/packages/inputmethods/LatinIME/+/master/dictionaries/fr_wordlist.combined.gz
The [changes](https://github.com/AnySoftKeyboard/LanguagePack/pull/47) will probably land in F-Droid and Play Store in the next days. Can you check the pack and tell whether this issue is solved and can be closed?

**Comment by breversa** [comment]
Great news, but why not use my proposed dictionary data (I can freshen them up if needed) ?

My rationale is that Dicollecte is a specialized project focused on french language, while for AOSP, french is just one language among many others. I thus expect the overall quality of Dicollecte's data to be better (if only slightly, or even the same, but in no way worse) than AOSP. :-)

**Comment by friesenkiwi** [comment]
There's nothing that speaks against the proposed Dicollecte data! :-) (As long as the License is compatible?) I was just in the general process of resurrecting all the old language packs, uplift them to the latest master branch/repository layout/build process, preparing them for release in F-Droid and Play Store and using the AOSP dictionaries as a default baseline for all of them. It's more like now providing the general possibility of working on the language packs here, by having them available as a up-to-date-branch.

So, yes, as soon as @menny merged the PR, please do refresh the Dicollecte data and add it to the build process, so that it gets merged with the AOSP data (or, if you are reasonably sure, just replace it). There is now a guideline on how to work in the pack: https://github.com/AnySoftKeyboard/LanguagePack/blob/master/README.md
Probably, you would need the `MakeDictionaryTask`, `GenerateWordsListTask` and `MergeWordsListTask`, for more information see here: https://github.com/AnySoftKeyboard/AnySoftKeyboardTools/blob/master/README.md
You can then put up a PR and the new and shiny pack with this new dictionary data will soon be available in an Android App store near you ;-)

**Comment by friesenkiwi** [comment]
Following up on the discussion over at https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/109
@breversa @papjul let's concentrate the discussion about a new French Dictionary in this issue here at the LanguagePack repo.
I would propose to merge https://github.com/AnySoftKeyboard/LanguagePack/pull/47 to have a common baseline but we don't need to release it as a pack rightaway. You can then work on incorporating the Dicollecte dictionary or try out the AOSP one first.
What Do you think about this?

There was an issue with the GenerateWordsListFromAOSPTask which didn't recognize the umlauts in the German dictionary. That was fixed with https://github.com/AnySoftKeyboard/AnySoftKeyboardTools/commit/9e1dc0b6ececc7548a4bd29b0798d4f155152519 two weeks ago, so it should now also show accented characters in French. Maybe try again?

The 'apostrophe case' (the original issue) seems to be, that apostrophe ' and hyphen - should be added to the list of word separators in the language pack, which is doable relatively easy.
There seem to be a couple of special cases drawn up by @parawizard, but that's only theoretical, until we have a working current common baseline pack.

@papjul which is the "another issue for long pressed characters" you are waiting to be fixed?

A Canadian pack with both English and French keyboard and dictionaries sounds interesting, but I'm not sure whether a single mixed/merged Dictionary would work out well, same goes for a Belgian pack with French and Dutch keyboard/dictionary?

**Comment by papjul** [comment]
I'm sorry but my pack is a start over from the beginning so I can't take old SVN pack as a base because I did some major changes like making common typographic symbols and common accentuated letters on keyboard more accessible for usage with French languages. But to make all of them correctly accessible I need one issue of ASK to be solved: AnySoftKeyboard/AnySoftKeyboard#707 and AnySoftKeyboard/AnySoftKeyboard#712

Thank you for pointing out the fix, I will try again the AOSP dictionary with this!

No, there is no issue with the hyphen on my side, I can have "peut-être" suggested correctly, in AOSP dictionary as well as Dicollecte dictionary because they are part of the dictionary. The only big issue is the apostrophe making writing in French with ASK a real pain. That's also why I made a dedicated key for the apostrophe, which reduces the pain.

While waiting for support for two dictionaries, I suggest we can merge AOSP dictionaries, like English+French if the fix works well, but you're right we can also make a French+Dutch dictionary. If in the end, it makes the package too big, we can put it aside for now. What do you think?

Not only Canadian and French/Belgian, I also plan to support QWERTZ from Swiss and BÉPO (I did some work on this but they don't comply yet with "common typographic/accentued letters accessible" I told you about in the 1st paragraph so I commented them in the code).

Please wait 'til I get home this evening and I will upload the code. The repo will be accessible here: https://git.amenysta.net/papjul/AnySoftKeyboardLanguagePackFrench (I created the repo a month ago but wanted to wait for fixes from ASK before commiting, but some have been fixed and you want to work together, so I will share it as soon as I'm home)

**Comment by friesenkiwi** [comment]
Very well, I can wait, no need to hurry :-)
I seems your work is much more progressed than what I did, so we should use your approaches, I'm happy to scrap my PR. But for better maintainability, it would be best to have it based on the current master branch template. Since there are really only the .xml files and the build.grade making a difference, maybe you can have a look at the current template or the PR proposed by me and just replace the corresponding files with your versions?

I originally got involved with this because I only wanted to resurrect the sources to all the language packs only available on Play Store but not on F-Droid, since the sources were "lost" in the SVN archive, but were not maintainable or even buildable with current Android Studio. My approach is somewhat stepwise: First make the old source available here in GitHub, modernize the build, re-release on Play Store and F-Droid, so people can put up pull requests to the separate packs, then release new versions with new features.

**Comment by papjul** [comment]
Don't worry, my work is based on master template on Dec 28, 2016. I just need to backport some little commits you did after.
However, as for Gradle, like I told you I have multiple dictionaries and I can't use it for automatic build, since each dictionary creation in the raw folder would erase the previous dictionary before creating the second dictionary. So I generate them one by one and then comment the gradle tasks. You will see what I did on my repo in a few hours…
I would also be happy to have some French/Belgian AZERTY users and Canadian QWERTY French speakers to help me test so I can focus on xml keyboards. Your PR contained also BÉPO so I may add it to the list of keyboards available in the first test release to not have less than what was available before ;)

**Comment by breversa** [comment]
I'll gladly test the BÉPO layout, as it was the reason I went to ASK to
begin with ! :-D

2017-02-17 15:09 GMT+01:00 Julien Papasian <notifications@github.com>:

**Comment by papjul** [comment]
I noticed that the BÉPO keyboard included in the 2011 version of French language pack has some keys missing (such as è or à) because it uses 10-keys per row instead of 11. I don't really see the point of using a BÉPO if you don't make use of è or à letters directly accessible, but I acknowledge that some people have big fingers so they prefer 10 keys per row, so I will implement both versions. To reduce the work to be done for first experimental build, which one would you like to test first?
See the map for reference: https://download.tuxfamily.org/dvorak/wiki/images/Carte-bepo-complete.png

**Comment by breversa** [comment]
I have a rather big-screened device and not-to-big fingers, so I guess I
can handle the 11-key version.

BTW, you *might* want to have a look at the Typematrix layout :
https://download.tuxfamily.org/dvorak/wiki/images/BepoTMparFabien.svg

2017-02-17 15:30 GMT+01:00 Julien Papasian <notifications@github.com>:

**Comment by papjul** [comment]
The Ç has enough space to be put on the 2nd row so I don't need to put it on the third row like TypeMatrix.
Also, I need to keep some space on the 3rd row for the Ê (that is missing on the TypeMatrix version).
As for the common typographic symbols keys . , and ' they are accessible on the left and right of the space bar (for all keyboard layouts, not just for BÉPO).

**Comment by papjul** [comment]
I pushed my code to https://git.amenysta.net/papjul/AnySoftKeyboardLanguagePackFrench
Dicollecte dictionary is the default dictionary and we will only test with this for now.
You need to change the bottom bar to French, because it is meant to.
Do you want me to build you a debug APK or can you compile it?

**Comment by breversa** [comment]
[lexique-dicollecte-fr-v6.0.2.zip](https://github.com/AnySoftKeyboard/LanguagePack/files/788142/lexique-dicollecte-fr-v6.0.2.zip)

There :
1- I updated my AWK script to work with Dicollecte's lexicon 6.0.2
2- I updated the fr.xml file accordingly

Feel free to update your APK with it as I don't know anything about Android development…

Note to self :
Never forget to dos2unix text files before working with them…

**Comment by papjul** [comment]
Don't worry, my dictionary is already up-to-date.
Here's my Python script: https://git.amenysta.net/papjul/AnySoftKeyboardLanguagePackFrench/src/master/dictionary/Dicollecte2Dictionary.py

**Comment by friesenkiwi** [comment]
OK, what are the next steps from your points of view?
Is the idea still to create a PR which can be merged after #47?

Also, maybe it would be a cool idea to reform either the awk or the python script into a Groovy Task (see https://github.com/AnySoftKeyboard/AnySoftKeyboardTools/tree/master/generatewordslist/src/main/groovy/com/anysoftkeyboard/tools/generatewordslist) which can then be used by Gradle during build, so everything can run automatically from the raw dicollecte file?

**Comment by papjul** [comment]
Next steps are tests by French-speakers to see that there are not regressions compared to the current French pack.
No, when did we say that? My repo is complete, have a look at: https://git.amenysta.net/papjul/AnySoftKeyboardLanguagePackFrench

Yes, I thought about a Groovy Task, however I don't know how to write in Groovy. I don't think it would be too hard to convert my Python script if one of you have some time and know how to do it.

**Comment by friesenkiwi** [comment]
OK, so I closed my PR.
But it would be really good, if you could rebase against the current branch at https://github.com/AnySoftKeyboard/LanguagePack/tree/French. It seems you started a new repo from scratch and only copied the files of the template/master branch over?
To have French speakers test it, we need a release apk. If you upload it here or provide it in another way, I can put it in my private F-Droid repo at https://fdroid.denk-nach-mcfly.de/fdroid/repo/ so it may be easier to install for more people for testing.

**Comment by papjul** [comment]
I downloaded the master branch and created all keyboards/dictionaries from scratch that I added to the repo. The branch you are showing me is even with master, so I don't know what you want me to rebase?
For next commits, I will cherry-pick commits from master branch occasionally to have a cleaner history if this is what you ask.

I can provide a debug APK for testing, and the release APK will then be signed by F-Droid, what do you think?

**Comment by friesenkiwi** [comment]
Your repository and the AnySoftKeyboard/LanguagePack currently are totally unrelated and independent from each other, speaking in Git terms.
In Git, the individual commits are identified by cryptographic hashes and the commits in a branch are chained by them signing each other, also incorporating merges. That's the way, how for example https://github.com/AnySoftKeyboard/LanguagePack/network can be computed.
Since your repository is not a fork or a branch of any of the branches of the AnySoftKeyboard/LanguagePack repository, they are not related. Because of that, a PR would most probably result in a conflict which would needed to be resolved manually.
To avoid this, you could clone the French branch (or, right now, since they are even, the master branch) locally on your computer. Cloning means to not only copy the current, latest state (like when downloading), but instead clone the complete repository with history, including all branches. You can push this to your git.amenysta.net and later push it to GitHub. Because of the cryptographic hashes, GitHub will recognize the history and be able to automatically merge it into the main repo.

Yes, of course, when the LanguagePack is released through the official F-Droid repository, it will be signed by their key, and the Play Store release will be signed by @menny s key. But if you want to provide a testing APK to people who have not enabled development settings on their device, you can't distribute a debug APK because it won't be installable. The debug/release setting is a flag in the .apk file. You can easily generate an own key for test releases, if we distribute it through my F-Droid repo (only an idea, might be easier for some people) it will be re-signed automatically anyways.

**Comment by papjul** [comment]
Fixed: https://git.amenysta.net/papjul/AnySoftKeyboardLanguagePackFrench/commits/master
OK, this is my first Android project, so I will have a look at the signing process documentation.

**Comment by friesenkiwi** [comment]
Nice, thank you! :-) :+1: 

In Android Studio, you can just do Build->Generate Signed APK... and then create a new Keystore in the dialog.

**Comment by friesenkiwi** [comment]
@papjul @breversa any news on this? Did any problems turn out during testing? If not, could you push your changes to a fork here at GitHub and put up a Pull Request?

**Comment by papjul** [comment]
I’ve been busy recently, but I will try to sign the APK tonight or this week-end.

**Comment by papjul** [comment]
Testers, could you please try this release build and tell me if you find issues with the 3 main keyboards or dictionary? https://amenysta.net/AnySoftKeyboardLanguagePackFrench-release-20170217.apk
Please also report which keyboard you use, so I’m sure we have testers for each keyboard.
Thanks!

**Comment by friesenkiwi** [comment]
Any news on this? Shouldn't we just release what is there currently? Fixes can be incorporated quite easily later.

**Comment by papjul** [comment]
I haven’t had any feedback from any testers, sorry :/
Yes it is a release APK, you can use it.

**Comment by breversa** [comment]
My bad, @papjul ! I got carried away in other things and totally forgot to test your language pack ! :-(

But there, I just installed and configured it ! I'll test it starting from now ! :-D
(For the record, I was using SwiftKey until then)

**Comment by breversa** [comment]
So, here are my first impressions after less than a day of use :

The good :
* Having "à" (see the ugly, though) and "e" in direct access is great :-)
* Grouping sentence-ending punctuation on the same key is great
* Having "…" as a single character is great ! :-)
* Having all accents for each letter is great

The bad :
* I have no use for the different apostrophes ; I'd rather have ":" and ";" instead
* Coming from a predictive keyboard app (SwiftKey) to a non-predictive on is hard… Prediction makes typing so much easier and faster.
* Words suggestions to the left of the screen is awkward in some cases : I have a rather big screen with small hands, so I use the "crammed to the right-side" option (I'm right-handed). Suggestions to the left force me to move my finger to the other side of the screen. Left-handed people might like it though ! SwiftKey deals with this by having the most probable suggestion in the middle of the screen, with less probable suggestions to both the left and right
* Words suggestions are a bit weird : sometimes, I feel they don't quite take frequency into account ; or maybe too many words have the same frequency ?

The ugly :
* Contraction of words with apostrophe are not part of the dictionnary : "c'est", "j'admire", "s'était", "s'apercevoir", "m'orienter", "t'abandonner", "n'avoir", etc.
* Typing "à" results in "et" being suggested ! And with "space = accept suggestion", this makes a big mess. What's more, for some reason, I can't add "à" to the dictionnary : tapping on it does not work !

**Comment by papjul** [comment]
Thank you for your feedback, could you contact me through https://amenysta.net/contact so we can exchange by email? If possible, could you prepare some screenshots related to what you're talking about?

> I have no use for the different apostrophes ; I'd rather have ":" and ";" instead. Or better : group them with ",", and put the smileys as long-press on the apostrophe key
: is directly accessible from V key, and ; is the second one (this is not as common as : in French).
’ is the official typographic symbol for contraction in French so we need it, though with computing, most people adopted the simple straight symbol '

**Comment by papjul** [comment]
Could you also force using Dicollecte dictionary by using long press > Substituer le dictionnaire par défaut on bottom right key? There might be some old dictionary data from another old French language pack because I can't seem to reproduce some of your words issues.

**Comment by friesenkiwi** [comment]
@papjul I pulled in your commits at https://github.com/friesenkiwi/LanguagePack/tree/french-papjul so it should later be easy to put up a PR to the "official" branch at https://github.com/AnySoftKeyboard/LanguagePack/tree/French ;-)

**Comment by Kravcian** [comment]
Hello Papjul, 

Thank you for your apk, I downloaded today and I'll give you a feedback within a week. I'll be using the bépo accents keyboard.

**Comment by xavihernandez** [comment]
Hi @papjul, how to explicitly specify "Dicollecte" dictionary? When I hit "Substituer le dictionnaire par défaut" I have 3 choices : "Par défaut", "Anglais" et "French".
Is "French" the good choice?

Apart that , as @breversa said, I really miss autocompletion for words with apostropha. "C'est", "j'ai", etc. are not proposed by suggestions. We arte oblige to: write word without apostropha, and force it (otherwise it gets corrected), or manually hit apostropha char to insert it.

I haven't been able to test it for too much time, as I was doing too much errors without apostropha handling.

**Comment by papjul** [comment]
Hi @xavihernandez, you are not testing my keyboard, my keyboard has two French dictionaries and one English+French dictionary (I may have removed some buggy ones). It can be downloaded here https://amenysta.net/AnySoftKeyboardLanguagePackFrench-release-20170217.apk
If this does not solve your issue, please uninstall the outdated French language pack if you have it installed.

The apostrophe issue is a well-known issue and not language pack related, please discuss in #109 for more info. Honestly, I don't use ASK because of this issue (and most French speaking users probably don't use it for the same reason, which is sad for the 8th most spoken language). At the beginning, people were using ASK for its Bépo keyboard, but now that modern keyboards support Bépo, only pure librist+bépo users remain and suffer from this issue. I hope that one day this issue will get in the top-priority buglist.

If you choose to disable autocompletion, please also go to Interface Utilisateur > Encore plus… >Rangée du bas > Français optimisé for best usage of the apostrophe.

**Comment by xavihernandez** [comment]
@papjul thanks for clarification. 
About apostrophe case, it works when you choose AOSP dictionary in ASK ("Substituer le dictionnaire par défaut"). I have attached a sceenshot.
![screenshot_20170607-130248](https://user-images.githubusercontent.com/3322064/26875348-b9b80eb0-4b81-11e7-95a8-c18a37eb5c10.png)

**Comment by Kravcian** [comment]
Hello @papjul 

Again thank you for your apk.

Here is my feedback, I'm comparing your apk with the ASK French "original" package that I downloaded on F-Droid. I used the Bépo keyboard.

What I liked : 
 - é, è, ê, à and ç are easely accessible
 
What I didn't like : 
 - "-" doesn't have a proper key on the principal keyboard
 - no automatic space between "!" or "?" or ";" and the previous word
- too many keys on the principal keyboard : many accents have their own keys but the keys are smaller to fit the screen; My suggestion to reduce the number of keys while keeping the accents easely accessible is the "double tap". For instance "è" could be the double tap of "é", "ê" of "e", "à" of "a", "ç" of "c", "ù" of "u", "ï" of "i", and even "!" of "?".

Thank you for your attention.

**Comment by papjul** [comment]
- "-" doesn't have a proper key on the principal keyboard => Hold the C letter, you will have many dashes (including typographic) available on main keyboard
- no automatic space between "!" or "?" or ";" and the previous word => I will have a look on how I can create an "espace fine insécable" but I cannot guarantee it is possible with ASK
- too many keys on the principal keyboard : many accents have their own keys but the keys are smaller to fit the screen; My suggestion to reduce the number of keys while keeping the accents easely accessible is the "double tap". For instance "è" could be the double tap of "é", "ê" of "e", "à" of "a", "ç" of "c", "ù" of "u", "ï" of "i", and even "!" of "?". => It is intended that this keyboard is 11-key large, there is another 10-key keyboard in the works but not for this first version, it was requested on Feb 17, see messages above. I'm interested by this double tap feature, on which ASK keyboard did you see it?

**Comment by Kravcian** [comment]
Hello, thank youfor your quick answer

Before I downloaded your apk, I used the bépo keyboard of the ASK French package available on F-Droid : https://f-droid.org/repository/browse/?fdfilter=french&fdid=com.anysoftkeyboard.languagepack.french_xlarge

- _no automatic space between "!" or "?" or ";" and the previous word => I will have a look on how I can create an "espace fine insécable" but I cannot guarantee it is possible with ASK_ ==> I think it is possible because it is implemented on the package I used before yours.

- _too many keys on the principal keyboard : many accents have their own keys but the keys are smaller to fit the screen; My suggestion to reduce the number of keys while keeping the accents easely accessible is the "double tap". For instance "è" could be the double tap of "é", "ê" of "e", "à" of "a", "ç" of "c", "ù" of "u", "ï" of "i", and even "!" of "?". => It is intended that this keyboard is 11-key large, there is another 10-key keyboard in the works but not for this first version, it was requested on Feb 17, see messages above. I'm interested by this double tap feature, on which ASK keyboard did you see it?_ ==> it is partially  implemented on the package I used before yours (only "è", "ù", "à" and "!")

**Comment by friesenkiwi** [comment]
Hey @papjul :-)
what's the status on this one? Are you satisfied with the current state and can it be released now?
I understand, that there might still be room for improvement and also that there is still https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/109 to be solved. But wouldn't the current development state still be a great improvement over the packs currently distributed via F-Droid and Play Store? So I suggest to first do a release, then get the suggestions, reports, reviews and comments from the community, work them in, then do another release.

@menny do you think, there is a chance to fix https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/109 in the foreseeable future?

CC @Kravcian @xavihernandez @breversa

**Comment by papjul** [comment]
Hi, I'm not even sure anymore about AnySoftKeyboard/AnySoftKeyboard#109 since it looks like it works fine with AOSP dictionary so it would mean there's no way we can use the Dicollecte dictionary (which is sad because this one is better). However, as a French native speaker, I noticed that in AOSP keyboard, the most used contractions in language are recognized, but the least common (however valid) with less used tenses (conjugation) will not be recognized so for people using autosuggestion it will be a pain in these cases. AnySoftKeyboard/AnySoftKeyboard#109 would help by recognizing for example that « m' » (contraction of « me ») then « aider » are two separate words, but you can't allow to use contraction with every word that begins with a voyel so it's actually more complicate.
I'm pretty satisfied for a first release, as there are indeed quite some improvements, but I think I should default to AOSP dictionary before doing a proper release due to the reports above (not sure when I will have time to do it). However, notice that the current Bépo keyboard is a 11-key large while current is 10-key so some people might not like this update (I noticed that one tester was doing more mistakes with it due to smaller keys).

**Comment by friesenkiwi** [comment]
OK, thank you, so we are looking forward to a PR soon! :-)

About the 109: That sounds weird, if it works with the AOSP dictionary, it should be possible to make it work with the Dicollecte one as well?! Maybe it's just a matter of a special character to use (replace in Dicollecte as a pre-task) or something like that?
About the Bépo, maybe you could first keep the old 10-key one as an additional keyboard for "conservative" people to select?

**Comment by papjul** [comment]
I think the AOSP dictionary considers « c’est » as a single word instead of ce + est, that's why it is suggested by one dictionary and not the other one. However, that's not how a dictionary is supposed to work (one word = one entry).
I designed a specific optimized layout for long-pressed keys so it would not be coherent to have just this one with an old different layout. I will note this on my todo-list but if you want it quick, it won’t be tested by other people.
I also remember that I did not fully tested the optimized bottom bar. You know, it changes according to input type: number, email address, URL, etc. But I don’t think it’s a showstopper.

**Comment by friesenkiwi** [comment]
Yes, I'd say let's release the current state quickly, even if some things might not be too thoroughly tested. It will still be a huge improvement over the old version.

**Comment by lillet23** [comment]
Could you please release this new version of the French keyboard?

**Comment by thiswillbeyourgithub** [comment]
Yeah I'm on v1.8-r12 from fdroid and it's *almost barely* usable in french :/

**Comment by dieudo4** [comment]
Does anyone know why there's no French APK indicated for F-Droid on this page ? :
* http://anysoftkeyboard.github.io/

**Comment by thiswillbeyourgithub** [comment]
@dieudo4 no idea

**Comment by milouse** [comment]
The apk was moved to the archive because of known vulnerabilities https://f-droid.org/wiki/page/AntiFeature:KnownVuln

You'll find a recent beta (but usable) version sooner in this thread. Some people are currently working on a new and totally rewritten french version. No release date has been announced yet. See @papjul messages.

**Comment by papjul** [comment]
I'm not using ASK anymore but you can find my latest work here that you can fork: https://git.amenysta.net/papjul/AnySoftKeyboardLanguagePackFrench and to only download the latest APK here: https://amenysta.net/AnySoftKeyboardLanguagePackFrench-release-20170217.apk
Since there is no French keyboard/dictionary for ASK anymore in F-Droid, I think you should push to F-Droid my latest work even if in "beta" status.

**Comment by dieudo4** [comment]
Thanks to both of you. Merci : )
@papjul why aren't you using ASK anymore ?

**Comment by lillet23** [comment]
@papjul I'm using your apk for the French keyboard. Great work!
(Je suis moi-même francophone).
Can anyone push it to F-droid? (sorry, I'm a bit of a noob, I still can't do that myself).

**Comment by menny** [comment]
Guys, I want to push a new version of this language pack the Play store and Fdroid. But I'm not sure where I can get the latest, most up-to-date source. Help?
@friesenkiwi

**Comment by Lyokovic** [comment]
There is currently a PR for the French package (#106).

I do not really get how it is related to the previous work mentionned here and I'm a bit lost about the commits it contains, but at least it works for me.

Is anyone having some clarifications about this?

**Comment by djibux** [comment]
I have been using the apk shared by @papjul.  Il works great.

**Comment by friesenkiwi** [comment]
Closing this because #106 was merged

**Comment by xavihernandez** [comment]
@friesenkiwi Just tried the latest version just released @ F-Droid (v1.9-r1), but when I choose Optimized  French for bottom line, I cannot get emoji panel. It just does not show on long press on comma key. It was working before the update.

**Comment by xavihernandez** [comment]
Hi any news about apostropha?
